### ISSUE TITLE
NullReferenceException in output window when showing output from Wakatime

### ISSUE THREAD

**Issue by jakubsuchybio** [wakatime/visualstudio-wakatime#52]
Exact error:
```
[Wakatime HandledException 04:00:21 dop.] GetPathFromMicrosoftRegister:: System.NullReferenceException: Object reference not set to an instance of an object.
   at WakaTime.PythonManager.GetPathFromMicrosoftRegister()
```

**Comment by alanhamlett** [comment]
Is this error preventing WakaTime from logging your coding activity?

**Comment by gandarez** [comment]
@jakubsuchybio Did you install python manually ou automatically by waka?

**Comment by jakubsuchybio** [comment]
@alanhamlett No. I see some activity on web dashboard.
@gandarez I don't remember, but I think I might install it manually.
Here are versions of pythons and waka extension that I have:
Python 3.5.0 (32-bit)   3.5.150.0
Python Tools 2.2.1 for Visual Studio 2015   2.2.31105.00
WakaTime extension 7.0.0

**Comment by gandarez** [comment]
The exception doesn't make sense at all. 
There's a try/catch block to deal with any exception inside that method.

```C#
static string GetPathFromMicrosoftRegister()
{
    try
    {
        var regex = new Regex(@"""([^""]*)\\([^""\\]+(?:\.[^"".\\]+))""");
        var pythonKey = Registry.ClassesRoot.OpenSubKey(@"Python.File\shell\open\command");
        var python = pythonKey.GetValue(null).ToString();
        var match = regex.Match(python);

        if (!match.Success) return null;

        var directory = match.Groups[1].Value;
        var fullPath = Path.Combine(directory, "pythonw");
        var process = new RunProcess(fullPath, "--version");

        process.Run();

        if (!process.Success)
            return null;

        Logger.Debug(string.Format("Python found from Microsoft Registery: {0}", fullPath));

        return fullPath;
    }
    catch (Exception ex)
    {
        Logger.Error("GetPathFromMicrosoftRegister:", ex);
        return null;
    }
}
```

Figuring out how you got it.

**Comment by jakubsuchybio** [comment]
Ok I downloaded source of WakaTime, tried to debug it and this line gives me null:
```
var pythonKey = Registry.ClassesRoot.OpenSubKey(@"Python.File\shell\open\command");
```

**Comment by jakubsuchybio** [comment]
So I made some more digging. In registry I didn't find Python.File sub-key for root folder. I then went into 'Programs and Features' found python, hit repair. Repair was successful. Then I looked into registry, the sub-key was there. And error doesn't show up.
However... I found that you have more fail-through methods to find python's path and this one is just one of them. So this failed with error, but some other found it and that's why the extension worked fine.

I would suggest you to add information into this Error in PythonManager.cs:58 to guide user to repair its Python installation.

**Comment by gandarez** [comment]
@jakubsuchybio Ok I agree with you but I tried to debug exactly as you done and didn't get the same result. Take a look at the screenshots below:
Forced NullReferenceException to be thrown
![image](https://cloud.githubusercontent.com/assets/782854/11565128/bace9f38-99c3-11e5-89d4-80cdc0262ab2.png)

Caught it
![image](https://cloud.githubusercontent.com/assets/782854/11565136/c170b66e-99c3-11e5-88b6-23192096b649.png)

Waka's log output pane
```[Wakatime HandledException 01:40:42 ] GetPathFromMicrosoftRegister:: System.NullReferenceException: Object reference not set to an instance of an object.
   at WakaTime.PythonManager.GetPathFromMicrosoftRegister() in c:\git\wakatime\visualstudio-wakatime\PythonManager.cs:line 38
[Wakatime Info 01:41:04 ] Initializing WakaTime v7.0.0
[Wakatime Info 01:41:05 ] Current wakatime-cli version is 4.1.8
[Wakatime Info 01:41:06 ] Checking for updates to wakatime-cli...
[Wakatime Info 01:41:06 ] wakatime-cli is up to date.
[Wakatime Info 01:41:06 ] Finished initializing WakaTime v7.0.0```
![image](https://cloud.githubusercontent.com/assets/782854/11565194/f9506016-99c3-11e5-893c-553622d10659.png)

**Comment by jakubsuchybio** [comment]
Well you get the same result, just with additional information about line that caused it. And that is because you have pdb file present when you were debugging it. When you install WakaTime into VS from binary, there are no pdb files present for the exception to have information where the problem occured exactly. Only the name of method. It should be like this if my memory about debugging from school is correct. :)

**Comment by gandarez** [comment]
@jakubsuchybio your're correct. What I mean is that I couldn't get any exception threw outisde the scope. However I'll did few mods into code to handle null reference as well.

**Comment by gandarez** [comment]
[Wakatime Warning 02:11:59 ] Couldn't find python's path through Microsft Register. Please try repairing your Python installation.

**Comment by gandarez** [comment]
@alanhamlett please compile it and upload to the gallery. Tks!

**Comment by jakubsuchybio** [comment]
Nice. I didn't mean exception out of scope. Just that output error, for clarification.

**Comment by gandarez** [comment]
I missed out, sorry!

**Comment by alanhamlett** [comment]
Released v7.0.1 with this fix. Thanks @gandarez and @jakubsuchybio!
